txtorcon/torcontrolprotocol.py

```diff
@@ -128,20 +129,18 @@ def got_update(self, data):
         # re-constitute these things like this. NS events are probably
         # all routers, so around 20000 lines.
 
-        # XXX at the very least, use StringIO or something to avoid
-        # all the string-concatenations
-
+        response_data = StringIO()
         lines = data.split('\n')
         if len(lines) > 1:
-            response_data = "650-" + self.name + "\r\n"
+            response_data.write("650-" + self.name + "\r\n")
             for line in range(len(lines)-1):
```

Comment:
looking good, but can you eliminate these +'s as well since we've got a stringio in here now?